src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -261,6 +278,35 @@ public async Task<IActionResult> Index(BulkActionViewModel model)
 
                         break;
 
+                    case nameof(TenantsBulkAction.Remove):
+                        if (String.Equals(shellSettings.Name, ShellHelper.DefaultShellName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            await _notifier.WarningAsync(H["You cannot remove the default tenant."]);
+                        }
+                        else if (shellSettings.State != TenantState.Disabled && shellSettings.State != TenantState.Uninitialized)
+                        {
+                            await _notifier.WarningAsync(H["The tenant '{0}' should be disabled or uninitialized.", shellSettings.Name]);
+                        }
+                        else
+                        {
+                            var context = await _shellRemovalManager.RemoveAsync(shellSettings);
+                            if (!context.Success)
+                            {
+                                await _notifier.ErrorAsync(H["An error occurred while removing the tenant '{0}'. {1}", shellSettings.Name, context.ErrorMessage]);
+                            }
+                            else
+                            {
+                                if (_logger.IsEnabled(LogLevel.Warning))
```

Comment:
Is this check is required here? I thought the service will write logs the logs file based on the logging configuration.

Reply:
Here we don't call the service, we just log that a controller action wanted to remove an already removed tenant. No problem to remove this log but I think @Piedone asked me to add it but not sure.  Hmm, it is in the bulk action section that we may want to remove.

Reply:
I perhaps asked for the logging but not for this `IsEnabled` check. The latter I don't think is necessary, since `_logger.LogWarning()` will only take an affect for those loggers that already are configured to log warnings.

Reply:
Okay will remove it

Reply:
Done